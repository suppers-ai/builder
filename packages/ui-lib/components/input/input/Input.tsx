import { InputProps } from "./Input.schema.ts";

export function Input({
  class: className = "",
  type = "text",
  size = "md",
  color,
  placeholder = "",
  value,
  bordered = true,
  ghost = false,
  disabled = false,
  id,
  min,
  max,
  step,
  autoComplete,
  showPasswordToggle = false,
  showColorPreview = true,
  onChange,
  onFocus,
  onBlur,
  onInput,
  // Extract schema-specific props that conflict with HTML attributes
  children,
  name,
  defaultValue,
  required,
  // Extract accessibility props to avoid type conflicts
  role,
  tabIndex,
  "aria-label": ariaLabel,
  "aria-describedby": ariaDescribedby,
  "aria-expanded": ariaExpanded,
  "aria-hidden": ariaHidden,
  ...props
}: InputProps) {
  // Set default placeholders based on input type
  const getDefaultPlaceholder = () => {
    if (placeholder) return placeholder;
    switch (type) {
      case "email": return "your@email.com";
      case "password": return "••••••••";
      case "number": return "0";
      case "date": return "";
      case "time": return "";
      case "color": return "";
      default: return "";
    }
  };

  // Set default autocomplete based on input type
  const getAutoComplete = () => {
    if (autoComplete) return autoComplete;
    switch (type) {
      case "email": return "email";
      case "password": return "current-password";
      default: return undefined;
    }
  };

  // Build input classes with type-specific adjustments
  const inputClasses = [
    "input",
    size ? `input-${size}` : "",
    color ? `input-${color}` : "",
    bordered ? "input-bordered" : "",
    ghost ? "input-ghost" : "",
    // Add padding for special input types with controls
    (type === "password" && showPasswordToggle) && "pr-12",
    (type === "number") && "pr-8",
    (type === "color" && showColorPreview) && "pl-12",
    className,
  ].filter(Boolean).join(" ");

  // For simple input types that don't need special UI
  const isSimpleInput = ["text", "email", "date", "time", "datetime-local", "tel", "url", "search"].includes(type);
  
  if (isSimpleInput) {
    return (
      <input
        type={type}
        class={inputClasses}
        placeholder={getDefaultPlaceholder()}
        value={value}
        disabled={disabled}
        id={id}
        name={name}
        defaultValue={defaultValue}
        required={required}
        min={min}
        max={max}
        step={step}
        autoComplete={getAutoComplete()}
        onChange={onChange}
        onFocus={onFocus}
        onBlur={onBlur}
        onInput={onInput}
        {...props}
      />
    );
  }

  // Password input with optional toggle (for islands/client-side components)
  if (type === "password") {
    if (showPasswordToggle) {
      // Return a wrapper div that can be enhanced with client-side functionality
      return (
        <div className="relative flex items-center">
          <input
            type="password"
            class={`${inputClasses} flex-1`}
            placeholder={getDefaultPlaceholder()}
            value={value}
            disabled={disabled}
            id={id}
            name={name}
            defaultValue={defaultValue}
            required={required}
            autoComplete={getAutoComplete()}
            onChange={onChange}
            onFocus={onFocus}
            onBlur={onBlur}
            onInput={onInput}
            {...props}
          />
          <button
            type="button"
            className="ml-2 p-2 text-gray-400 hover:text-gray-600 transition-colors"
            disabled={disabled}
            aria-label="Toggle password visibility"
            data-password-toggle="true"
          >
            <svg className="w-5 h-5" fill="none" stroke="currentColor" viewBox="0 0 24 24">
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
              />
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                d="M2.458 12C3.732 7.943 7.523 5 12 5c4.478 0 8.268 2.943 9.542 7-1.274 4.057-5.064 7-9.542 7-4.477 0-8.268-2.943-9.542-7z"
              />
            </svg>
          </button>
        </div>
      );
    } else {
      // Simple password input without toggle
      return (
        <input
          type="password"
          class={inputClasses}
          placeholder={getDefaultPlaceholder()}
          value={value}
          disabled={disabled}
          id={id}
          name={name}
          defaultValue={defaultValue}
          required={required}
          autoComplete={getAutoComplete()}
          onChange={onChange}
          onFocus={onFocus}
          onBlur={onBlur}
          onInput={onInput}
          {...props}
        />
      );
    }
  }

  // Number input with increment/decrement controls
  if (type === "number") {
    const stepValue = step || 1; // Default step for number inputs
    
    const increment = () => {
      const currentValue = (value as number) || 0;
      const newValue = currentValue + (stepValue as number);
      if (max === undefined || newValue <= (max as number)) {
        const event = new Event("change");
        Object.defineProperty(event, "target", {
          value: { value: newValue.toString() },
          enumerable: true,
        });
        onChange?.(event);
      }
    };

    const decrement = () => {
      const currentValue = (value as number) || 0;
      const newValue = currentValue - (stepValue as number);
      if (min === undefined || newValue >= (min as number)) {
        const event = new Event("change");
        Object.defineProperty(event, "target", {
          value: { value: newValue.toString() },
          enumerable: true,
        });
        onChange?.(event);
      }
    };

    return (
      <div className="relative">
        <input
          type="number"
          class={inputClasses}
          placeholder={getDefaultPlaceholder()}
          value={value}
          disabled={disabled}
          id={id}
          name={name}
          defaultValue={defaultValue}
          required={required}
          min={min}
          max={max}
          step={stepValue}
          onChange={onChange}
          onFocus={onFocus}
          onBlur={onBlur}
          onInput={onInput}
          {...props}
        />
        <div className="absolute right-1 top-1/2 -translate-y-1/2 flex flex-col">
          <button
            type="button"
            className="text-xs text-base-content/50 hover:text-base-content disabled:cursor-not-allowed"
            onClick={increment}
            disabled={disabled || (max !== undefined && ((value as number) || 0) >= (max as number))}
          >
            ▲
          </button>
          <button
            type="button"
            className="text-xs text-base-content/50 hover:text-base-content disabled:cursor-not-allowed"
            onClick={decrement}
            disabled={disabled || (min !== undefined && ((value as number) || 0) <= (min as number))}
          >
            ▼
          </button>
        </div>
      </div>
    );
  }

  // Color input with preview
  if (type === "color") {
    return (
      <div className="relative">
        {showColorPreview && (
          <div
            className="absolute left-2 top-1/2 -translate-y-1/2 w-6 h-6 rounded border border-base-300"
            style={{ backgroundColor: (value as string) || "#000000" }}
          />
        )}
        <input
          type="color"
          class={inputClasses}
          value={(value as string) || "#000000"}
          disabled={disabled}
          id={id}
          name={name}
          defaultValue={defaultValue}
          required={required}
          onChange={onChange}
          onFocus={onFocus}
          onBlur={onBlur}
          onInput={onInput}
          {...props}
        />
      </div>
    );
  }

  // Fallback for any other input types
  return (
    <input
      type={type as string}
      class={inputClasses}
      placeholder={getDefaultPlaceholder()}
      value={value}
      disabled={disabled}
      id={id}
      name={name}
      defaultValue={defaultValue}
      required={required}
      min={min}
      max={max}
      step={step}
      autoComplete={getAutoComplete()}
      onChange={onChange}
      onFocus={onFocus}
      onBlur={onBlur}
      onInput={onInput}
      {...props}
    />
  );
}
