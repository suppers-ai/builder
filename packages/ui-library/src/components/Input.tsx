// Fresh 2.0 island component
import { useSignal } from "@preact/signals";
import { ComponentEventHandlers, ThemedComponentProps } from '../types.ts';

export interface InputProps extends ThemedComponentProps, ComponentEventHandlers {
  type?: 'text' | 'email' | 'password' | 'number' | 'tel' | 'url' | 'search';
  placeholder?: string;
  value?: string;
  defaultValue?: string;
  disabled?: boolean;
  readonly?: boolean;
  required?: boolean;
  name?: string;
  autoComplete?: string;
  autoFocus?: boolean;
  maxLength?: number;
  minLength?: number;
  pattern?: string;
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  variant?: 'default' | 'error' | 'success';
  fullWidth?: boolean;
  label?: string;
  helperText?: string;
  errorText?: string;
  leftIcon?: string;
  rightIcon?: string;
}

export default function Input({
  type = 'text',
  placeholder,
  value,
  defaultValue,
  disabled = false,
  readonly = false,
  required = false,
  name,
  autoComplete,
  autoFocus = false,
  maxLength,
  minLength,
  pattern,
  size = 'md',
  variant = 'default',
  fullWidth = false,
  label,
  helperText,
  errorText,
  leftIcon,
  rightIcon,
  className = '',
  style = {},
  onChange,
  onFocus,
  onBlur,
  onKeyDown,
  onKeyUp,
  id,
  'data-testid': testId,
  ...props
}: InputProps) {
  // Fresh 2.0 island state management
  const isFocused = useSignal(false);
  const inputValue = useSignal(value || defaultValue || '');
  
  const inputId = id || `input-${Math.random().toString(36).substr(2, 9)}`;
  const hasError = variant === 'error' || !!errorText;
  const hasSuccess = variant === 'success';

  const baseClasses = [
    'block',
    'border',
    'rounded-md',
    'shadow-sm',
    'transition-colors',
    'focus:outline-none',
    'focus:ring-2',
    'focus:ring-offset-2',
    'disabled:opacity-50',
    'disabled:cursor-not-allowed',
    'read-only:bg-gray-50',
    'read-only:cursor-default',
  ];

  const variantClasses = {
    default: [
      'border-gray-300',
      'focus:border-blue-500',
      'focus:ring-blue-500',
    ],
    error: [
      'border-red-300',
      'focus:border-red-500',
      'focus:ring-red-500',
    ],
    success: [
      'border-green-300',
      'focus:border-green-500',
      'focus:ring-green-500',
    ],
  };

  const sizeClasses = {
    xs: ['px-2', 'py-1', 'text-xs'],
    sm: ['px-3', 'py-1.5', 'text-sm'],
    md: ['px-3', 'py-2', 'text-sm'],
    lg: ['px-4', 'py-2', 'text-base'],
    xl: ['px-4', 'py-3', 'text-base'],
  };

  const inputClasses = [
    ...baseClasses,
    ...variantClasses[hasError ? 'error' : hasSuccess ? 'success' : 'default'],
    ...sizeClasses[size],
    fullWidth ? 'w-full' : '',
    leftIcon ? 'pl-10' : '',
    rightIcon ? 'pr-10' : '',
    className,
  ].filter(Boolean).join(' ');

  const containerClasses = [
    fullWidth ? 'w-full' : '',
    'relative',
  ].filter(Boolean).join(' ');

  // Fresh 2.0 island event handlers with state management
  const handleFocus = (event: FocusEvent) => {
    isFocused.value = true;
    onFocus?.(event);
  };

  const handleBlur = (event: FocusEvent) => {
    isFocused.value = false;
    onBlur?.(event);
  };

  const handleChange = (event: Event) => {
    const target = event.target as HTMLInputElement;
    inputValue.value = target.value;
    onChange?.(event);
  };

  // Dynamic classes based on island state
  const dynamicInputClasses = [
    ...baseClasses,
    ...variantClasses[hasError ? 'error' : hasSuccess ? 'success' : 'default'],
    ...sizeClasses[size],
    fullWidth ? 'w-full' : '',
    leftIcon ? 'pl-10' : '',
    rightIcon ? 'pr-10' : '',
    isFocused.value ? 'ring-2 ring-offset-2' : '',
    className,
  ].filter(Boolean).join(' ');

  return (
    <div className={containerClasses}>
      {label && (
        <label
          htmlFor={inputId}
          className="block text-sm font-medium text-gray-700 mb-1"
        >
          {label}
          {required && <span className="text-red-500 ml-1">*</span>}
        </label>
      )}
      
      <div className="relative">
        {leftIcon && (
          <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
            <span className="text-gray-400 text-sm">{leftIcon}</span>
          </div>
        )}
        
        <input
          id={inputId}
          type={type}
          name={name}
          placeholder={placeholder}
          value={inputValue.value}
          disabled={disabled}
          readOnly={readonly}
          required={required}
          autoComplete={autoComplete}
          autoFocus={autoFocus}
          maxLength={maxLength}
          minLength={minLength}
          pattern={pattern}
          className={dynamicInputClasses}
          style={style}
          data-testid={testId}
          onChange={handleChange}
          onFocus={handleFocus}
          onBlur={handleBlur}
          onKeyDown={onKeyDown}
          onKeyUp={onKeyUp}
          {...props}
        />
        
        {rightIcon && (
          <div className="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none">
            <span className="text-gray-400 text-sm">{rightIcon}</span>
          </div>
        )}
      </div>
      
      {(helperText || errorText) && (
        <p className={`mt-1 text-xs ${hasError ? 'text-red-600' : 'text-gray-500'}`}>
          {errorText || helperText}
        </p>
      )}
    </div>
  );
}