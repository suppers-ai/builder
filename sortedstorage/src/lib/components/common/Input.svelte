<script lang="ts">
	import type { ComponentType } from 'svelte';
	
	export let type: 'text' | 'email' | 'password' | 'number' | 'tel' | 'url' | 'search' = 'text';
	export let value: string | number = '';
	export let placeholder = '';
	export let label = '';
	export let error = '';
	export let disabled = false;
	export let required = false;
	export let readonly = false;
	export let icon: ComponentType | null = null;
	export let id = `input-${Math.random().toString(36).substr(2, 9)}`;
	
	let className = '';
	export { className as class };
	
	$: inputClass = `
		w-full px-3 py-2 
		border rounded-lg 
		focus:outline-none focus:ring-2 focus:ring-primary-500
		transition-colors
		${icon ? 'pl-10' : ''}
		${error 
			? 'border-red-500 focus:ring-red-500' 
			: 'border-gray-300 dark:border-gray-600'}
		${disabled 
			? 'bg-gray-100 dark:bg-gray-800 opacity-75 cursor-not-allowed' 
			: 'bg-white dark:bg-gray-700'}
		${className}
	`.trim();
</script>

<div class="w-full">
	{#if label}
		<label for={id} class="block text-sm font-medium mb-2">
			{label}
			{#if required}
				<span class="text-red-500">*</span>
			{/if}
		</label>
	{/if}
	
	<div class="relative">
		{#if icon}
			<div class="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400">
				<svelte:component this={icon} class="w-5 h-5" />
			</div>
		{/if}
		
		<input
			{id}
			{type}
			{value}
			{placeholder}
			{disabled}
			{required}
			{readonly}
			class={inputClass}
			on:input
			on:change
			on:blur
			on:focus
			on:keydown
			on:keyup
		/>
	</div>
	
	{#if error}
		<p class="mt-1 text-sm text-red-500">{error}</p>
	{/if}
</div>