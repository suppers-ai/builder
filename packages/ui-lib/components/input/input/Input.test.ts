import { assertEquals, assertStringIncludes } from "https://deno.land/std@0.224.0/assert/mod.ts";
import { assertSnapshot } from "https://deno.land/std@0.224.0/testing/snapshot.ts";
import { renderToString } from "preact-render-to-string";
import { Input } from "./Input.tsx";

Deno.test("Input - basic rendering", () => {
  const html = renderToString(Input({}));
  assertStringIncludes(html, "input");
  assertStringIncludes(html, "input-md");
  assertStringIncludes(html, "input-bordered");
  assertStringIncludes(html, 'type="text"');
});

Deno.test("Input - text type (default)", () => {
  const html = renderToString(Input({}));
  assertStringIncludes(html, 'type="text"');
});

Deno.test("Input - email type", () => {
  const html = renderToString(Input({
    type: "email",
  }));
  assertStringIncludes(html, 'type="email"');
});

Deno.test("Input - password type", () => {
  const html = renderToString(Input({
    type: "password",
  }));
  assertStringIncludes(html, 'type="password"');
});

Deno.test("Input - number type", () => {
  const html = renderToString(Input({
    type: "number",
  }));
  assertStringIncludes(html, 'type="number"');
});

Deno.test("Input - with placeholder", () => {
  const html = renderToString(Input({
    placeholder: "Enter your name",
  }));
  assertStringIncludes(html, 'placeholder="Enter your name"');
});

Deno.test("Input - with value", () => {
  const html = renderToString(Input({
    value: "Test value",
  }));
  assertStringIncludes(html, 'value="Test value"');
});

Deno.test("Input - disabled", () => {
  const html = renderToString(Input({
    disabled: true,
  }));
  assertStringIncludes(html, "disabled");
});

Deno.test("Input - enabled (default)", () => {
  const html = renderToString(Input({
    disabled: false,
  }));
  assertEquals(html.includes("disabled"), false);
});

Deno.test("Input - bordered (default)", () => {
  const html = renderToString(Input({}));
  assertStringIncludes(html, "input-bordered");
});

Deno.test("Input - not bordered", () => {
  const html = renderToString(Input({
    bordered: false,
  }));
  assertEquals(html.includes("input-bordered"), false);
});

Deno.test("Input - ghost variant", () => {
  const html = renderToString(Input({
    ghost: true,
  }));
  assertStringIncludes(html, "input-ghost");
});

Deno.test("Input - not ghost (default)", () => {
  const html = renderToString(Input({
    ghost: false,
  }));
  assertEquals(html.includes("input-ghost"), false);
});

Deno.test("Input - size variants", () => {
  const sizes = ["xs", "sm", "md", "lg"];

  sizes.forEach((size) => {
    const html = renderToString(Input({
      size: size as any,
    }));
    assertStringIncludes(html, `input-${size}`);
  });
});

Deno.test("Input - color variants", () => {
  const colors = ["primary", "secondary", "accent", "info", "success", "warning", "error"];

  colors.forEach((color) => {
    const html = renderToString(Input({
      color: color as any,
    }));
    assertStringIncludes(html, `input-${color}`);
  });
});

Deno.test("Input - without color", () => {
  const html = renderToString(Input({}));
  // Should not have any color class
  const hasColorClass = ["primary", "secondary", "accent", "info", "success", "warning", "error"]
    .some((color) => html.includes(`input-${color}`));
  assertEquals(hasColorClass, false);
});

Deno.test("Input - custom class", () => {
  const html = renderToString(Input({
    class: "custom-input-class",
  }));
  assertStringIncludes(html, "custom-input-class");
});

Deno.test("Input - with id", () => {
  const html = renderToString(Input({
    id: "test-input",
  }));
  assertStringIncludes(html, 'id="test-input"');
});

Deno.test("Input - all props combined", () => {
  const html = renderToString(Input({
    type: "email",
    placeholder: "Enter your email",
    value: "test@example.com",
    size: "lg",
    color: "primary",
    bordered: true,
    ghost: false,
    disabled: false,
    class: "test-class",
    id: "full-input",
  }));
  assertStringIncludes(html, 'type="email"');
  assertStringIncludes(html, 'placeholder="Enter your email"');
  assertStringIncludes(html, 'value="test@example.com"');
  assertStringIncludes(html, "input-lg");
  assertStringIncludes(html, "input-primary");
  assertStringIncludes(html, "input-bordered");
  assertEquals(html.includes("input-ghost"), false);
  assertEquals(html.includes("disabled"), false);
  assertStringIncludes(html, "test-class");
  assertStringIncludes(html, 'id="full-input"');
});

Deno.test("Input - email input with primary color", () => {
  const html = renderToString(Input({
    type: "email",
    color: "primary",
    placeholder: "email@example.com",
  }));
  assertStringIncludes(html, 'type="email"');
  assertStringIncludes(html, "input-primary");
  assertStringIncludes(html, 'placeholder="email@example.com"');
});

Deno.test("Input - password input with secondary color", () => {
  const html = renderToString(Input({
    type: "password",
    color: "secondary",
    placeholder: "Enter password",
  }));
  assertStringIncludes(html, 'type="password"');
  assertStringIncludes(html, "input-secondary");
  assertStringIncludes(html, 'placeholder="Enter password"');
});

Deno.test("Input - small ghost input", () => {
  const html = renderToString(Input({
    size: "sm",
    ghost: true,
    bordered: false,
  }));
  assertStringIncludes(html, "input-sm");
  assertStringIncludes(html, "input-ghost");
  assertEquals(html.includes("input-bordered"), false);
});

Deno.test("Input - large error input", () => {
  const html = renderToString(Input({
    size: "lg",
    color: "error",
    placeholder: "Invalid input",
  }));
  assertStringIncludes(html, "input-lg");
  assertStringIncludes(html, "input-error");
  assertStringIncludes(html, 'placeholder="Invalid input"');
});

Deno.test("Input - disabled with value", () => {
  const html = renderToString(Input({
    disabled: true,
    value: "Disabled value",
  }));
  assertStringIncludes(html, "disabled");
  assertStringIncludes(html, 'value="Disabled value"');
});

Deno.test("Input - number input", () => {
  const html = renderToString(Input({
    type: "number",
    placeholder: "Enter a number",
    value: "42",
  }));
  assertStringIncludes(html, 'type="number"');
  assertStringIncludes(html, 'placeholder="Enter a number"');
  assertStringIncludes(html, 'value="42"');
});

Deno.test("Input - search input", () => {
  const html = renderToString(Input({
    type: "search",
    placeholder: "Search...",
    color: "accent",
  }));
  assertStringIncludes(html, 'type="search"');
  assertStringIncludes(html, 'placeholder="Search..."');
  assertStringIncludes(html, "input-accent");
});

Deno.test("Input - tel input", () => {
  const html = renderToString(Input({
    type: "tel",
    placeholder: "+1 (555) 123-4567",
  }));
  assertStringIncludes(html, 'type="tel"');
  assertStringIncludes(html, 'placeholder="+1 (555) 123-4567"');
});

Deno.test("Input - url input", () => {
  const html = renderToString(Input({
    type: "url",
    placeholder: "https://example.com",
  }));
  assertStringIncludes(html, 'type="url"');
  assertStringIncludes(html, 'placeholder="https://example.com"');
});

// Snapshot tests
Deno.test("Input - HTML snapshot basic", async (t) => {
  const html = renderToString(Input({}));
  await assertSnapshot(t, html);
});

Deno.test("Input - HTML snapshot with all props", async (t) => {
  const html = renderToString(Input({
    type: "email",
    placeholder: "Enter your email address",
    value: "user@example.com",
    size: "lg",
    color: "primary",
    id: "email-input",
  }));
  await assertSnapshot(t, html);
});

Deno.test("Input - HTML snapshot variants", async (t) => {
  const inputs = [
    Input({ type: "text", placeholder: "Text input", size: "sm" }),
    Input({ type: "email", placeholder: "Email input", color: "primary" }),
    Input({ type: "password", placeholder: "Password input", color: "secondary" }),
    Input({ type: "number", placeholder: "Number input", size: "lg" }),
    Input({ ghost: true, placeholder: "Ghost input" }),
    Input({ disabled: true, value: "Disabled input" }),
  ];

  const html = inputs.map((input) => renderToString(input)).join("\n");
  await assertSnapshot(t, html);
});

Deno.test("Input - HTML snapshot sizes", async (t) => {
  const sizes = ["xs", "sm", "md", "lg"];
  const htmls = sizes.map((size) =>
    renderToString(Input({
      size: size as any,
      placeholder: `Size ${size}`,
      color: "primary",
    }))
  );
  await assertSnapshot(t, htmls.join("\n"));
});

Deno.test("Input - HTML snapshot colors", async (t) => {
  const colors = ["primary", "secondary", "accent", "info", "success", "warning", "error"];
  const htmls = colors.map((color) =>
    renderToString(Input({
      color: color as any,
      placeholder: `Color ${color}`,
      size: "md",
    }))
  );
  await assertSnapshot(t, htmls.join("\n"));
});
