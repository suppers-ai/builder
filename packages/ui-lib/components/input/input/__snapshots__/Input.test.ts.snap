export const snapshot = {};

snapshot[`Input - HTML snapshot basic 1`] = `'<input type="text" class="input input-md input-bordered" placeholder/>'`;

snapshot[`Input - HTML snapshot with all props 1`] = `'<input type="email" class="input input-lg input-primary input-bordered" placeholder="Enter your email address" value="user@example.com" id="email-input" autocomplete="email"/>'`;

snapshot[`Input - HTML snapshot variants 1`] = `
'<input type="text" class="input input-sm input-bordered" placeholder="Text input"/>
<input type="email" class="input input-md input-primary input-bordered" placeholder="Email input" autocomplete="email"/>
<input type="password" class="input input-md input-secondary input-bordered" placeholder="Password input" autocomplete="current-password"/>
<div class="relative"><input type="number" class="input input-lg input-bordered pr-8" placeholder="Number input" step="1"/><div class="absolute right-1 top-1/2 -translate-y-1/2 flex flex-col"><button type="button" class="text-xs text-base-content/50 hover:text-base-content disabled:cursor-not-allowed">▲</button><button type="button" class="text-xs text-base-content/50 hover:text-base-content disabled:cursor-not-allowed">▼</button></div></div>
<input type="text" class="input input-md input-bordered input-ghost" placeholder="Ghost input"/>
<input type="text" class="input input-md input-bordered" placeholder value="Disabled input" disabled/>'
`;

snapshot[`Input - HTML snapshot sizes 1`] = `
'<input type="text" class="input input-xs input-primary input-bordered" placeholder="Size xs"/>
<input type="text" class="input input-sm input-primary input-bordered" placeholder="Size sm"/>
<input type="text" class="input input-md input-primary input-bordered" placeholder="Size md"/>
<input type="text" class="input input-lg input-primary input-bordered" placeholder="Size lg"/>'
`;

snapshot[`Input - HTML snapshot colors 1`] = `
'<input type="text" class="input input-md input-primary input-bordered" placeholder="Color primary"/>
<input type="text" class="input input-md input-secondary input-bordered" placeholder="Color secondary"/>
<input type="text" class="input input-md input-accent input-bordered" placeholder="Color accent"/>
<input type="text" class="input input-md input-info input-bordered" placeholder="Color info"/>
<input type="text" class="input input-md input-success input-bordered" placeholder="Color success"/>
<input type="text" class="input input-md input-warning input-bordered" placeholder="Color warning"/>
<input type="text" class="input input-md input-error input-bordered" placeholder="Color error"/>'
`;
